### PR TITLE
Add docs on common stacks best practices

### PR DESCRIPTION
**Pull request by strickvl**
Focused on the AWS and GCP sections since for the rest we have a lot of other best practices in the component docs (which those guides already point back to). 

Also not quite sure how much of this is redundant.

### CODE REVIEW

**Review by htahir1 — COMMENTED** [review]
Maybe we can do this part of the restructuring but I thought it might make sense to feature this a bit more prominently ? However in general this looks good I’ll take a look tomorrow again

File: docs/book/how-to/popular-integrations/aws-guide.md [hunk 1]
```diff
@@ -286,6 +286,90 @@ Now that you have a functional AWS stack set up with ZenML, you can explore more
...
+## Best Practices for Using an AWS Stack with ZenML
+
+When working with an AWS stack in ZenML, consider the following best practices
+to optimize your workflow, enhance security, and improve cost-efficiency. These
+are all things you might want to do or amend in your own setup once you have
+tried running some pipelines on your AWS stack.
+
+### Use IAM Roles and Least Privilege Principle
+
+Always adhere to the principle of least privilege when setting up IAM roles. Only grant the minimum permissions necessary for your ZenML pipelines to function. Regularly review and audit your IAM roles to ensure they remain appropriate and secure.
```
Comment (wjayesh) [inline docs/book/how-to/popular-integrations/aws-guide.md:302]:
```suggestion
Always adhere to the principle of least privilege when setting up IAM roles. Only grant the minimum permissions necessary for your ZenML pipelines to function. Regularly review and audit your [IAM roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html) to ensure they remain appropriate and secure.
```

File: docs/book/how-to/popular-integrations/aws-guide.md [hunk 2]
```diff
@@ -286,6 +286,90 @@ Now that you have a functional AWS stack set up with ZenML, you can explore more
...
+
+### Leverage AWS Resource Tagging
+
+Implement a consistent tagging strategy for your AWS resources. To tag an S3 bucket:
```
Comment (wjayesh) [inline docs/book/how-to/popular-integrations/aws-guide.md:306]:
```suggestion
Implement a [consistent tagging strategy](https://aws.amazon.com/solutions/guidance/tagging-on-aws/) for all of your AWS resources that you use for your pipelines. For example, if you have S3 as an artifact store in your stack, you should tag it like shown below:

```

File: docs/book/how-to/popular-integrations/aws-guide.md [hunk 3]
```diff
@@ -286,6 +286,90 @@ Now that you have a functional AWS stack set up with ZenML, you can explore more
...
+```shell
+aws s3api put-bucket-tagging --bucket your-bucket-name --tagging 'TagSet=[{Key=Project,Value=ZenML},{Key=Environment,Value=Production}]'
+```
+
+These tags will help you with billing and cost allocation tracking and also with
+any cleanup efforts.
+
+### Implement Cost Management Strategies
+
+Use AWS Cost Explorer and AWS Budgets to monitor and manage your spending. To create a cost budget:
```
Comment (wjayesh) [inline docs/book/how-to/popular-integrations/aws-guide.md:317]:
```suggestion
Use [AWS Cost Explorer](https://aws.amazon.com/aws-cost-management/aws-cost-explorer/) and [AWS Budgets](https://aws.amazon.com/aws-cost-management/aws-budgets/) to monitor and manage your spending. To create a cost budget:
```

File: docs/book/how-to/popular-integrations/aws-guide.md [hunk 4]
```diff
@@ -286,6 +286,90 @@ Now that you have a functional AWS stack set up with ZenML, you can explore more
...
+2. Create the cost budget:
+
+```shell
+aws budgets create-budget --account-id your-account-id --budget file://budget-config.json
+```
+
+Set up cost allocation tags to track expenses related to your ZenML projects:
+
+```shell
+aws ce create-cost-category --name ZenML-Projects --rules-version 1 --rules file://rules.json
```
Comment (wjayesh) [inline docs/book/how-to/popular-integrations/aws-guide.md:352]: did you check if this is correct? i think it should be [create-cost-category-definition](https://docs.aws.amazon.com/cli/latest/reference/ce/create-cost-category-definition.html)

File: docs/book/how-to/popular-integrations/aws-guide.md [hunk 5]
```diff
@@ -286,6 +286,90 @@ Now that you have a functional AWS stack set up with ZenML, you can explore more
...
+    keep_alive_period_in_seconds = 300, # 5 minutes, default value
+)
+```
+
+This configuration keeps instances warm for 5 minutes after each job completes, allowing subsequent jobs to start faster if initiated within this timeframe. The reduced startup time can be particularly beneficial for iterative development processes or frequently run pipelines.
+
+### Implement a Robust Backup Strategy
+
+Regularly backup your critical data and configurations. For S3, enable versioning and consider using cross-region replication for disaster recovery.
+
```
Comment (wjayesh) [inline docs/book/how-to/popular-integrations/aws-guide.md:372]: links here too

File: docs/book/how-to/popular-integrations/gcp-guide.md [hunk 6]
```diff
@@ -175,4 +175,68 @@ If you do not want to use any of the created resources in the future, simply del
...
+
+Always adhere to the principle of least privilege when setting up IAM roles. Only grant the minimum permissions necessary for your ZenML pipelines to function. Regularly review and audit your IAM roles to ensure they remain appropriate and secure.
+
+### Leverage GCP Resource Labeling
+
+Implement a consistent labeling strategy for your GCP resources. To label a GCS
+bucket, for example:
+
+```shell
+gsutil label set project-labels.json gs://your-bucket-name
```
Comment (wjayesh) [inline docs/book/how-to/popular-integrations/gcp-guide.md:192]: gsutil is not the recommended CLI. use gcloud storage instead. see here: https://cloud.google.com/storage/docs/using-bucket-labels#set-label-cli

File: docs/book/how-to/popular-integrations/gcp-guide.md [hunk 7]
```diff
@@ -175,4 +175,68 @@ If you do not want to use any of the created resources in the future, simply del
...
+  "project": "zenml",
+  "environment": "production"
+}
+```
+
+These labels will help you with billing and cost allocation tracking and also with any cleanup efforts.
+
+### Implement Cost Management Strategies
+
+Use Google Cloud's Cost Management tools to monitor and manage your spending. To set up a budget alert:
```
Comment (wjayesh) [inline docs/book/how-to/popular-integrations/gcp-guide.md:208]: links here

**Review by wjayesh — CHANGES_REQUESTED** [review]
in general, it needs more links (i have added some already) and some fixes to the commands. good work thinking about the headings, not much more i would add to that

File: docs/book/how-to/popular-integrations/aws-guide.md (hunk 4 above)
Comment (strickvl) [reply docs/book/how-to/popular-integrations/aws-guide.md:353]: you're right!

File: docs/book/how-to/popular-integrations/gcp-guide.md (hunk 6 above)
Comment (strickvl) [reply docs/book/how-to/popular-integrations/gcp-guide.md:192]: updated now

**Review by htahir1 — APPROVED** [review]
I think this is fine for now, lets maybe make it more prominent in the restructuring

**Review by wjayesh — APPROVED** [review]

**Pull request merged**